src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp

```diff
@@ -163,6 +163,41 @@ static FlatSymbolRefAttr getOrInsertMemcpy(
   return SymbolRefAttr::get(context, "llvm.memcpy.p0i8.p0i8.i64");
 }
 
+static FlatSymbolRefAttr getOrInsertRandomNormal(
+    PatternRewriter &rewriter, ModuleOp module, Type inType) {
+  MLIRContext *context = module.getContext();
+  StringRef functionName = inType.isF64() ? "get_random_normal_value_f64"
+                                          : "get_random_normal_value_f32";
+  if (module.lookupSymbol<LLVM::LLVMFuncOp>(functionName.str()))
+    return SymbolRefAttr::get(context, functionName.str());
+
+  // Signature of the input is:
+  //  "krnl.random_normal"(%0, %c60, %cst, %cst_0, %cst_1)
+  // with types:
+  //  (memref<3x4x5xf32>, index, f32, f32, f32)
+  // or
+  //  (memref<3x4x5xf64>, index, f64, f64, f64)
+  auto llvmVoidTy = LLVM::LLVMVoidType::get(context);
+  auto llvmOptionsTy = FloatType::getF32(context);
+  auto llvmOutputTy = LLVM::LLVMPointerType::get(llvmOptionsTy);
+  if (inType.isF64()) {
+    llvmOptionsTy = FloatType::getF64(context);
+    llvmOutputTy = LLVM::LLVMPointerType::get(llvmOptionsTy);
+  }
+  auto llvmI64Ty = IntegerType::get(context, 64);
+  auto llvmFnType = LLVM::LLVMFunctionType::get(llvmVoidTy,
+      ArrayRef<mlir::Type>({llvmOutputTy, llvmI64Ty, llvmOptionsTy,
+          llvmOptionsTy, llvmOptionsTy}),
+      false);
+
+  // Insert the memcpy function into the body of the parent module.
```

Comment:
The comment mentions memcpy but you are creating the function type for "get_random_normal_value_f[64|32]".